File: sliding_window/sum.js
/**
 * Finds a contiguous subset of arr whose sum equals targetSum.
 * Returns the subset if found, otherwise returns -1.
 */
class Soluation{
    targetSum(arr,targetSum){
        var sum =0;
        let l=0;
        let r =0;
        let flag = false;
        for(r=0; r<arr.length; r++){
            sum = sum + arr[r];
            while(sum > targetSum){
                sum = sum - arr[l]
                l++;
            }
            if(sum == targetSum){
                flag = true;
                break
            }
        }
        // console.log("hello");
        if(flag){
            return arr.slice(l,r+1);

        }
        else{
            return -1;
        }
    }
}
var myInstance = new Soluation();
console.log(myInstance.targetSum([1,3,6,3,5,7],16));

File: sliding_window/sum.py
def target_sum(arr, target_sum):
        """
        Finds a contiguous subset of arr whose sum equals target_sum.
        Returns the subset if found, otherwise returns -1.
        """
        curr_sum = 0
        left = 0        
        for right in range(len(arr)):
            curr_sum += arr[right]            
            # Shrink window while sum is greater than target
            while curr_sum > target_sum:
                curr_sum -= arr[left]
                left += 1           
            # If we found the target sum, return the subarray
            if curr_sum == target_sum:
                return arr[left:right + 1]       
        return -1

test_arr = [1, 3, 6, 17, 4, 5, 7]
test_sum = 16
    
result = target_sum(test_arr, test_sum)
 
print(f"Result: {result}")
    
 
 
 